### ISSUE TITLE
[Bug]: sp-number-field cannot press increment/decrement buttons with VO on MacOS

### ISSUE THREAD

**Issue by jnurthen** [adobe/spectrum-web-components#3271]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-number-field

### Expected behavior

When I use VO cursor to move to the Increment/Decrement buttons I should be able to press them to modify the field

### Actual behavior

Nothing happens when I press the buttons using VO

### What browsers are you seeing the problem in?

Safari

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/number-field/
2. Go to any of the Number Fields in the page
3. Make sure VO is on
4. Press CTRL+OPTION+Right arrow to focus on the increment button
5. Press CTRL+OPTION+SPACE to press the button. 
Note the value does not increase

**Comment by najikahalsema** [comment]
Related to #3274 #3272

**Comment by susmithayenugula** [comment]
Hi team, 
I verified that this issue exists on Safari and not on Chrome. Observe the focus ring during VO in Safari:
<img width="151" alt="numberfield_VO_safari" src="https://github.com/adobe/spectrum-web-components/assets/21351513/86076a18-1486-4ddd-8282-0c3a414942b6">

On Chrome, the focus ring doesn't bleed beyond the boundaries of the button. 
<img width="172" alt="numberfield_VO_Chrome" src="https://github.com/adobe/spectrum-web-components/assets/21351513/6478ae52-a3cb-4fc1-9abe-573aeccbc47e">

I wonder if the boundary is causing the issue. The `sp-infield-button` has `aria-label` set with `role="button"`. I don't see any specific CSS for `outline` or `box-shadow`. I do see some `border` variants that I will look into.

Looking for pointers on how to investigate further, thank you!

**Comment by najikahalsema** [comment]
SWC-269

**Comment by caseyisonit** [comment]
Hello @jnurthen in order to help us triage your issue, [can you please review our bug severity classification guide](https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md#issue-severity-classification) and provide us with the appropriate severity classification? Please update the label to the correct severity level and remove `SEV missing` label.

If we don't hear from you in 2 weeks, we will close this issue as not planned.

<img width="1024" height="1028" alt="Image" src="https://github.com/user-attachments/assets/d5fd78ee-6e8e-4967-97f3-4e4198b79f13" />
